src/main/java/de/hpi/bpt/argos/storage/DatabaseAccessImpl.java

```diff
@@ -0,0 +1,235 @@
+		Transaction transaction = null;
+		try (Session session = sessionFactory.openSession()) {
+			transaction = session.beginTransaction();
+			int batchCount = 0;
+
+			for (PersistenceArtifact artifact : artifacts) {
+				session.saveOrUpdate(artifact);
+
+				if (++batchCount % sessionBatchSize == 0) {
+
+					logger.debug(String.format("saving artifacts... %1$d / %2$d", batchCount, artifacts.length));
+
+					session.flush();
+					session.clear();
+				}
+			}
+
+			transaction.commit();
+		} catch (Exception e) {
+			if (transaction != null) {
+				transaction.rollback();
+			}
+
+			LoggerUtilImpl.getInstance().error(logger, "cannot save/update artifacts in database", e);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean deleteArtifacts(PersistenceArtifact... artifacts) {
+		Transaction transaction = null;
+		try (Session session = sessionFactory.openSession()) {
+			transaction = session.beginTransaction();
+
+			for (PersistenceArtifact artifact : artifacts) {
+				session.delete(artifact);
+			}
+
+			transaction.commit();
+		} catch (Exception e) {
+			if (transaction != null) {
+				transaction.rollback();
+			}
+
+			LoggerUtilImpl.getInstance().error(logger, "cannot delete artifacts in database", e);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public <ResultType, QueryType> ResultType getArtifacts(Session session,
+														   Query<QueryType> query,
+														   Transaction transaction,
+														   Callable<ResultType> getValue,
+														   ResultType defaultValue) {
+
+		ObjectWrapper<ResultType> result = new ObjectWrapper<>();
+		WatchImpl.measure("get artifacts",
+				() -> result.set(executeGetArtifacts(session, query, transaction, getValue, defaultValue)));
+
+		return result.get();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public <ResultType, ResultImplType extends ResultType> List<ResultType> getArtifactsById(Session session,
+																							 Class<ResultImplType> resultTypeClass,
+																							 Long... ids) {
+
+		ObjectWrapper<List<ResultType>> result = new ObjectWrapper<>();
+		WatchImpl.measure("get artifacts by id",
+				() -> result.set(executeGetArtifactsById(session, resultTypeClass, ids)));
+		return result.get();
+	}
+
+	/**
+	 * This method actually executes the getArtifact-method.
+	 * @param session - the database session, which must be open
+	 * @param query - the query to execute and to retrieve the results from
+	 * @param transaction - the current transaction
+	 * @param getValue - the function to get the results from the query
+	 * @param defaultValue - a fall back default value in case anything went wrong or no entities were found
+	 * @param <ResultType> - the result type
+	 * @param <QueryType> - the query type
+	 * @return - an object of the result value type
+	 */
+	private <ResultType, QueryType> ResultType executeGetArtifacts(Session session,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)